File: src/main/java/se/osoco/api/model/BalanceReport.java
package se.osoco.api.model;

import java.util.List;

public class BalanceReport {
    private final List<Account> assets;
    private final List<Account> liabilitiesAndEquityCapital;


    public BalanceReport(se.osoco.domain.reports.BalanceSheet balanceSheet) {
        assets = balanceSheet.assets().totalAssets().accounts().stream().map(Account::new).sorted().toList();
        liabilitiesAndEquityCapital = balanceSheet.liabilitiesAndEquityCapital()
                .totalLiabilitiesAndEquityCapital().accounts().stream().map(Account::new).sorted().toList();
    }


}

File: src/main/client/src/domain/BalanceReport.ts
import Account from "./Account";

export default class BalanceReport {
  private _assets: Account[];
  private _liabilitiesAndEquityCapital: Account[];

  constructor(assets: Account[], liabilitiesAndEquityCapital: Account[]) {
    this._assets = assets;
    this._liabilitiesAndEquityCapital = liabilitiesAndEquityCapital;
  }

  public get assets(): Account[] {
    return this._assets;
  }

  public get liabilitiesAndEquityCapital(): Account[] {
    return this._liabilitiesAndEquityCapital;
  }
}
